v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -15,131 +19,98 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass
-class ReferenceDataGlobals:
+@dataclasses.dataclass
+class Globals:
     paths: dict[str]
     versions: dict[str]
     enums: dict[str, dict[str, list[str]]]
-
-    def __init__(self, globals_struct: hl.Struct):
-        self.paths = self._struct_to_dict(globals_struct.paths)
-        self.versions = self._struct_to_dict(globals_struct.versions)
-        self.enums = self._struct_to_dict(globals_struct.enums)
-
-    def _struct_to_dict(self, struct: hl.Struct) -> dict:
-        result_dict = {}
-        for field in struct:
-            if isinstance(struct[field], hl.Struct):
-                result_dict[field] = self._struct_to_dict(struct[field])
+    selects: dict[str, set[str]]
+
+    def __getitem__(self, name: str):
+        return getattr(self, name)
+
+    @classmethod
+    def from_dataset_configs(
+        cls,
+        rdc: ReferenceDatasetCollection,
+        dataset_type: DatasetType,
+        reference_genome: ReferenceGenome,
+    ):
+        paths, versions, enums, selects = {}, {}, {}, {}
+        for dataset in rdc.datasets(dataset_type):
+            dataset_config = CONFIG[dataset][reference_genome.v02_value]
+            dataset_ht = import_ht_from_config_path(dataset_config, reference_genome)
+
+            paths[dataset] = get_ht_path(dataset_config)
+            versions[dataset] = hl.eval(
+                parse_dataset_version(
+                    dataset_ht,
+                    dataset,
+                    dataset_config,
+                ),
+            )
+            enums[dataset] = dataset_config.get('enum_select', {})
+            selects[dataset] = set(
+                get_all_select_fields(dataset_ht, dataset_config).keys(),
+            )
+        return cls(paths, versions, enums, selects)
+
+    @classmethod
+    def from_ht(
+        cls,
+        ht: hl.Table,
+        rdc: ReferenceDatasetCollection,
+        dataset_type: DatasetType,
+    ):
+        rdc_globals_struct = hl.eval(ht.globals)
+        paths = dict(rdc_globals_struct.paths)
+        versions = dict(rdc_globals_struct.versions)
+        enums = dict(rdc_globals_struct.enums)
+
+        selects = {}
+        for dataset in rdc.datasets(dataset_type):
+            if dataset in ht.row:
+                selects[dataset] = set(ht[dataset])
+        return cls(paths, versions, enums, selects)
+
+
+class GlobalsValidator:
```

Comment:
I think this stands out to me as being a class here.  It doesn't look like we're doing anything "state"-ey with it (Maybe storing the {"dataset": result} somewhere is an option?) and we're just initializing it only to call a single method with no arguments.

Reply:
honestly yes the class is unnecessary I just made them functions again